### DISCUSSION TITLE
Publish the nightly version of the book

### DISCUSSION THREAD

**Discussion by azzamsa** [KDAB/cxx-qt#431] [Ideas] [answered]
Hi.

Is there a nightly version of the book? 
Dioxus had: 1) https://dioxuslabs.com/guide/ for stable version 2) https://dioxuslabs.com/nightly/guide/ for nightly version

The latter version helps people try out the `master/main` branch without digging into the book's git repository.

Today, I find myself digging into the code to find that `qt_module` changed to [qt_modules](https://github.com/KDAB/cxx-qt/blame/2c96be55d5e32183310725426d31a2f6c8d868b7/crates/cxx-qt-build/src/lib.rs#L307). Also, I can't find the changes in `unreleased` section of the [CHANGELOG.md](https://github.com/KDAB/cxx-qt/blob/main/CHANGELOG.md). So I need to make sure the relevant section is in the `book/` directory.

I think it would be helpful if we have 1) a nightly version of the book 2) any breaking changes listed in the CHANGELOG.md

Thank you for maintaining cxx-qt :heart:

**Comment by Be-ing** [comment, marked as answer]
This would be nice. The question is where to host it. Maybe we could use https://kdab.github.io/cxx-qt/book/ for the prerelease builds and move the book for the releases to a subdomain of https://kdab.com.

**Reply by ahayzen-kdab** [reply]
Before we wondered about using the github pages still and simply have `https://kdab.github.io/cxx-qt/book/stable/` and `https://kdab.github.io/cxx-qt/book/nightly/` or something ? As the action could build into the relevant sub folder `main` and the latest tagged release etc.

**Reply by ahayzen-kdab** [reply]
Also hopefully once we get closer to being stable this will become less of a problem :-)

**Reply by Be-ing** [reply]
I think publishing both to GitHub Pages would be possible, but more work to set up the CI.

**Comment by Be-ing** [comment]
In the meantime you can view prelease versions of the book by running `mdbook serve` from the `book` directory in the repository.
